propagation/propagation.go

```diff
@@ -40,6 +40,29 @@ type TextMapCarrier interface {
 	// must never be done outside of a new major release.
 }
 
+// BytesMapCarrier is a Carrier that stores propagated
+// values in a map of bytes.
+type BytesMapCarrier map[string][]byte
+
+// Get returns the value associated with the passed key.
+func (c BytesMapCarrier) Get(key string) string {
+	return string(c[key])
+}
+
+// Set stores the key-value pair.
+func (c BytesMapCarrier) Set(key string, value string) {
```

Comment:
Why input the value as a string and store it as a []byte?

Reply:
@jmacd  So that it implements the `TextMapCarrier` interface.

@rakyll Maybe it would be worth adding `var _ TextMapCarrier = (*BytesMapCarrier)(nil)` ?

Reply:
@rakyll would you mind following up, here?